collection-scripts/gather_performance_profile_controller

```diff
@@ -0,0 +1,162 @@
+
+POD_IP=$(hostname -I |  tr -d "[:blank:]" )
+echo "[$NAMESPACE/$POD_IP/$POD_NAME]" >> ${BASE_COLLECTION_PATH}/debug
+oc get pod -n $NAMESPACE $POD_NAME -o json >> ${BASE_COLLECTION_PATH}/debug
+
+sed -i -e "s#MUST_GATHER_IMAGE#$MUST_GATHER_IMAGE#" $DAEMONSET_MANIFEST
+
+oc create -f $NAMESPACE_MANIFEST
+oc create -f $SERVICEACCOUNT_MANIFEST
+oc adm policy add-scc-to-user privileged -n perf-node-gather -z perf-node-gather
+oc create -f $CLUSTER_ROLE_MANIFEST
+oc create -f $CLUSTER_ROLE_BINDING_MANIFEST
+oc create -f $DAEMONSET_MANIFEST
+
+COUNTER=0
+until check_node_gather_pods_ready || [ $COUNTER -eq 300 ]; do
+   (( COUNTER++ ))
+   sleep 1
+done
+
+for line in $(oc get pod -o=custom-columns=NODE:.spec.nodeName --no-headers --field-selector=status.phase!=Running -n perf-node-gather)
+do
+    echo "Failed to collect perf-node-gather data from node ${line} due to pod scheduling failure." >> ${NODES_PATH}/skipped_nodes.txt
+done
+
+COLLECTABLE_NODES=()
+for line in $(oc get pod -o=custom-columns=NODE:.spec.nodeName,NAME:.metadata.name --no-headers --field-selector=status.phase=Running -n perf-node-gather)
+do
+    node=$(echo $line | awk -F ' ' '{print $1}')
+    pod=$(echo $line | awk -F ' ' '{print $2}')
+    NODE_PATH=${NODES_PATH}/$node
+    mkdir -p "${NODE_PATH}"
+
+    oc exec $pod -n perf-node-gather -- lspci -nvv > $NODE_PATH/lspci
+    oc exec $pod -n perf-node-gather -- lscpu -e > $NODE_PATH/lscpu
+    oc exec $pod -n perf-node-gather -- cat /proc/cmdline > $NODE_PATH/proc_cmdline
+    oc exec $pod -n perf-node-gather -- dmesg > $NODE_PATH/dmesg
+    oc exec $pod -n perf-node-gather -- ethtool -k eth0 > $NODE_PATH/ethtool_features
+    oc exec $pod -n perf-node-gather -- ethtool -l eth0 > $NODE_PATH/ethtool_channels
+
+    COLLECTABLE_NODES+=($node)
+
+    oc exec $pod -n perf-node-gather -- gather_sysinfo --json cpuaff --procfs=/host/proc --sysfs=/host/sys > $NODE_PATH/cpu_affinities.json
+    oc exec $pod -n perf-node-gather -- gather_sysinfo --json irqaff --procfs=/host/proc --sysfs=/host/sys > $NODE_PATH/irq_affinities.json
+    oc exec $pod -n perf-node-gather -- gather_sysinfo --json podres --socket-path=unix:///host/podresources/kubelet.sock > $NODE_PATH/podresources.json
+
+    oc exec $pod -n perf-node-gather -- gather_sysinfo snapshot --debug --root=/host --output=- > $NODE_PATH/sysinfo.tgz 2> $NODE_PATH/sysinfo.log
+
+    oc exec $pod -n perf-node-gather -- gather_sysinfo podinfo --node-name $node > $NODE_PATH/pods_info.json
+done
+
+# Collect journal logs for specified units for all nodes
+NODE_UNITS=(kubelet)
+ADM_PIDS=()
+for NODE in ${COLLECTABLE_NODES[@]}; do
+    NODE_PATH=${NODES_PATH}/$NODE
+    mkdir -p ${NODE_PATH}
+    for UNIT in ${NODE_UNITS[@]}; do
+        timeout -k 5m 30m bash -c "oc adm node-logs $NODE -u $UNIT --since '-8h' | gzip" > ${NODE_PATH}/${NODE}_logs_$UNIT.gz &
+	ADM_PIDS+=($!)
+    done
+done
+wait "${ADM_PIDS[@]}"
+
+oc delete -f $DAEMONSET_MANIFEST
```

Comment:
I don't think we need to do this; if we rely on the invoking `oc` command to clean things up for us.

Reply:
for everyone?